### ISSUE TITLE
Arabic characters are not allowed 

### ISSUE THREAD

**Issue by nonmakina** [towards-a-new-leftypol/leftypol_lainchan#125]
Arabic text is removed >>35097 from posts due to this line in strip_combining_chars:
https://github.com/towards-a-new-leftypol/leftypol_lainchan/blob/5ef52b3c71763defc7d551ebd27157e98a0d89fb/inc/functions.php#L2267
> if ( ($ord >= 768 && $ord <= 879) || ($ord >= 1536 && $ord <= 1791) || ($ord >= 3655 && $ord <= 3659) || ($ord >= 7616 && $ord <= 7679) || ($ord >= 8400 && $ord <= 8447) || ($ord >= 65056 && $ord <= 65071))

The limits converted to hex:
[code]
>>> s = 'if ( ($ord >= 768 && $ord <= 879) || ($ord >= 1536 && $ord <= 1791) || ($ord >= 3655 && $ord <= 3659) || ($ord >= 7616 && $ord <= 7679) || ($ord >= 8400 && $ord <= 8447) || ($ord >= 65056 && $ord <= 65071))'
>>> import re
>>> ' '.join ('{:06x}'.format (int (mo.group ())) for mo in re.finditer ('[0-9]+', s))
'000300 00036f 000600 0006ff 000e47 000e4b 001dc0 001dff 0020d0 0020ff 00fe20 00fe2f'
[/code]

The relevant unicode blocks:
[code]
  6 000300 - 00036F    112 COMBINING_DIACRITICAL_MARKS
 12 000600 - 0006FF    256 ARABIC
 31 000E00 - 000E7F    128 THAI
 67 001DC0 - 001DFF     64 COMBINING_DIACRITICAL_MARKS_SUPPLEMENT
 73 0020D0 - 0020FF     48 COMBINING_MARKS_FOR_SYMBOLS
156 00FE20 - 00FE2F     16 COMBINING_HALF_MARKS
[/code]

Four COMBINING_* blocks are removed, along with five characters from THAI and the entire ARABIC block. The five THAI characters are:
[code]
U+0E47 Mn THAI CHARACTER MAITAIKHU
U+0E48 Mn THAI CHARACTER MAI EK
U+0E49 Mn THAI CHARACTER MAI THO
U+0E4A Mn THAI CHARACTER MAI TRI
U+0E4B Mn THAI CHARACTER MAI CHATTAWA
[/code]

However, strip_combining_chars is only applied to:
https://github.com/towards-a-new-leftypol/leftypol_lainchan/blob/5ef52b3c71763defc7d551ebd27157e98a0d89fb/post.php#L803
> $post['name'] = strip_combining_chars($post['name']);
> $post['email'] = strip_combining_chars($post['email']);
> $post['subject'] = strip_combining_chars($post['subject']);
> $post['body'] = strip_combining_chars($post['body']);
which means that leftypol already allows arabic characters in post file names, such as this post >>32563 in the webm thread.

If a decision is made to allow arabic characters in posts, this can be implemented by removing the arabic block limits
> ($ord >= 1536 && $ord <= 1791)
from strip_combining_chars.

The commit that stripped arabic characters is:
https://github.com/towards-a-new-leftypol/leftypol_lainchan/commit/126ee42b9dffe262457177c5dbb7ffa2db170763
> Author: czaks <marcin@6irc.net>
> Date:   Fri May 6 14:34:42 2016 +0200
> better rules for stripping combined chars, based on 45c0d327619 by @ctrlcctrlv

**Comment by ctrlcctrlv** [comment]
This is obviously wrong as to the Unicode Standard. (Yes I know it's my commit, people learn things over time. :wink:)

The correct way to strip combining characters is not to specify _codepoints_ matching, but rather Unicode _properties_. In particular, you want to strip characters with classes `Me` and `Mn`. This is also future-proof.

Stripping all Arabic is obviously dumb.

**Comment by dedushka1** [comment]
@ctrlcctrlv sorry to bother you, but do you think this might have to do with emoji getting corrupted when users try to edit their posts?

**Comment by ctrlcctrlv** [comment]
Sorry, that doesn't ring a bell for me and seems unrelated. However there might be some issue with non BMP codepoints somewhere

**Comment by discomrade** [comment]
> The correct way to strip combining characters is not to specify _codepoints_ matching, but rather Unicode _properties_. In particular, you want to strip characters with classes `Me` and `Mn`. This is also future-proof.

I've made a feature to allow a small number of sequential combining characters (e.g. 1-3 is fine, more gets stripped) to try and allow foreign languages but disallow malicious use.
Do combining characters present any security risk in small amounts, or was the stripping just to stop zalgo text interfering with other posts?

A review of this code would be greatly appreciated. Closing this issue since Arabic text can now be posted.
